libs/logging/src/export.ts

```diff
@@ -0,0 +1,122 @@
+        message: `Malformed log line identified, log line will be ignored: ${log} `,
+        result: 'Log line will not be included in CDF output',
+        disposition: 'failure',
+      });
+      continue;
+    }
+    const decodedLog = decodedLogResult.ok();
+    assert(typeof decodedLog['timeLogWritten'] === 'string'); // While this is not enforced in the LogLine type the zod schema will enforce it is always present so we know this to be true.
+
+    const rawDecodedObject = JSON.parse(log);
+    const customInformation = extractAdditionalKeysFromObj(
+      decodedLog,
+      rawDecodedObject
+    );
+
+    const standardDispositionResult = safeParse(
+      z.nativeEnum(EventLogging.EventDispositionType),
+      decodedLog.disposition
+    );
+    const disposition = standardDispositionResult.isOk()
+      ? standardDispositionResult.ok()
+      : decodedLog.disposition === ''
+      ? EventLogging.EventDispositionType.Na
+      : EventLogging.EventDispositionType.Other;
+    const cdfEvent: EventLogging.Event = {
+      '@type': 'EventLogging.Event',
+      Id: decodedLog.eventId,
+      Disposition: disposition,
+      OtherDisposition:
+        disposition === 'other' ? decodedLog.disposition : undefined,
+      Sequence: idx.toString(),
+      TimeStamp: decodedLog['timeLogWritten'],
+      Type: decodedLog.eventType,
+      Description: decodedLog.message,
+      Details: JSON.stringify({
+        ...customInformation,
+        source: decodedLog.source,
+      }),
+      UserId: decodedLog.user,
+    };
+    yield cdfEvent;
+  }
+}
+
+export async function* buildCdfLog(
+  logger: Logger,
+  electionDefinition: ElectionDefinition,
+  logFileReader: AsyncIterable<string>,
+  machineId: string,
+  codeVersion: string,
+  currentUser: LoggingUserRole
+): AsyncIterable<string> {
+  const source = logger.getSource();
+
+  if (!CLIENT_SIDE_LOG_SOURCES.includes(source)) {
+    await logger.log(LogEventId.LogConversionToCdfComplete, currentUser, {
+      message: 'The current application is not able to export logs.',
+      result: 'Log file not converted to CDF format.',
+      disposition: 'failure',
+    });
+    throw new Error('Can only export CDF logs from a frontend app.');
+  }
+
+  const currentDevice: JsonStreamInput<EventLogging.Device> = {
+    '@type': 'EventLogging.Device',
+    Type: DEVICE_TYPES_FOR_APP[source],
+    Id: machineId,
+    Version: codeVersion,
+    Event: generateCdfEventsForExport(logger, currentUser, logFileReader),
+  };
+  const eventElectionLog: JsonStreamInput<EventLogging.ElectionEventLog> = {
+    '@type': 'EventLogging.ElectionEventLog',
+    Device: [currentDevice],
+    ElectionId: electionDefinition.electionHash,
+    GeneratedTime: new Date().toISOString(),
+  };
+
+  await logger.log(LogEventId.LogConversionToCdfComplete, currentUser, {
+    message: 'Log file successfully converted to CDF format.',
+    disposition: 'success',
+  });
+
+  return yield* jsonStream(eventElectionLog);
```

Comment:
The other main change is that the output now streams as well as the input by using `jsonStream` instead of `JSON.stringify`.